utoipa-gen/tests/schema_derive_test.rs

```diff
@@ -3533,6 +3533,30 @@ fn derive_schema_with_custom_field_with_schema() {
     )
 }
 
+#[test]
+fn derive_unit_type() {
+    let data = api_doc! {
+        struct Data {
+            unit_type: ()
+        }
+    };
+
+    assert_json_eq!(
+        data,
+        json!({
+            "type": "object",
+            "required": [ "unit_type" ],
+            "properties": {
+                "unit_type": {
+                    "type": "object",
+                    "default": null,
+                    "nullable": true
+                }
+            }
+        })
+    )
+}
+
```

Comment:
This test is okay :+1: This is exactly what should be tested. Here we create `ToSchema` derived struct with unit type and see whether it creates correct json when tokenized with ToTokens in `component/schema.rs` file. 

And when introduced breaking changes these tests then will make sure that the implementation always stays correct, because after all the changes are done, the tests need to pass the build.